### PR TITLE
witness: init at 0.1.1

### PR DESCRIPTION
**Pull request by fkautz**
###### Motivation for this change
Adding a new package for cryptographically attesting a CI/CD process with in-toto.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/security/witness/default.nix [hunk 1]
```diff
@@ -0,0 +1,27 @@
+{ lib, buildGoModule, fetchFromGitHub }:
+
+buildGoModule rec {
+  pname = "witness";
+  version = "0.1.1";
+
+  outputs = [ "out" ];
+
```
Comment (zowoq) [inline pkgs/tools/security/witness/default.nix:8]: Default, not needed.
```suggestion
```

**Comment by fkautz** [comment]
I'm not sure if I should add anything to the 22.05 release notes. Please let me know if I should make an entry.

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -10851,6 +10851,10 @@ with pkgs;
     SDL = SDL_sixel;
   };
 
+  witness = callPackage ../tools/security/witness {
+    buildGoModule = buildGo117Module;
+  };
```
Comment (zowoq) [inline pkgs/top-level/all-packages.nix:10856]: `1.17` has recently been made the default.
```suggestion
  witness = callPackage ../tools/security/witness { };
```

**Review by zowoq — COMMENTED** [review]
> I'm not sure if I should add anything to the 22.05 release notes.

Just adding a package doesn't require a release note.

**Comment by fkautz** [comment]
I've merged the two suggested changes.

**Comment by zowoq** [comment]
Please squash the three package commits into one, I can't squash when merging as the maintainer commit should remain separate.

**Comment by fkautz** [comment]
I've squashed them.

File: pkgs/tools/security/witness/default.nix [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+  };
+
+  vendorSha256 = "sha256-zkLparWJsuqrhOQxxV37dBqt6fwpSinTO+paJkbl+sM=";
+
+  # We only want the witness binary, not the helper utilities for generating docs.
+  subPackages = [ "cmd/witness" ];
+
+  meta = with lib; {
+    description = "A pluggable framework for software supply chain security. Witness prevents tampering of build materials and verifies the integrity of the build process from source to target";
+    homepage = "github.com/testifysec/witness";
```
Comment (Synthetica9) [inline pkgs/tools/security/witness/default.nix:21]:
```suggestion
    homepage = "https://github.com/testifysec/witness";
```

**Comment by fkautz** [comment]
I've added the URL fix and squashed it.

**Pull request merged**